controllers/dtversion/docker_config.go

```diff
@@ -7,16 +7,17 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
-type DockerConfigAuth struct {
-	Username string
-	Password string
+type DockerConfig struct {
+	Auths         map[string]DockerAuth
+	SkipCertCheck bool
 }
 
-type DockerConfig struct {
-	Auths map[string]DockerConfigAuth
+type DockerAuth struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
 }
 
-func NewDockerConfig(secret *corev1.Secret) (*DockerConfig, error) {
+func ParseDockerConfigJSON(secret *corev1.Secret) (map[string]DockerAuth, error) {
```

Comment:
Function is not given a json string, but a secret. Also, it doesn't return a Docker config anymore, but the authentication configurations. renaming it to something akin to `ParseDockerAuthsFromSecret` might be better